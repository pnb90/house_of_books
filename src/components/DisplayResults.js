import React from 'react'
import styled from 'styled-components'
import Card from './Card'
import Slide from '@material-ui/core/Slide';

const DisplayContainer = styled.div`
  display: flex;
  width: 85%; 
  height: 500px;
  background: white;
  border-radius: 5px;
  flex-wrap: wrap;
  justify-content: center;
  padding: 20px;
  overflow: auto;
  background: #f1f8e9;
`

function DisplayResults(props) {
  var displayBooks = null

  if (props.books) {
    displayBooks = props.books.map(book => {
      return(
        <Slide>
          <Card 
            book = { book }
          />      
        </Slide>
      )})
  } else if (props === undefined ) {
    displayBooks = "Oh no - 0 search results. Try another search"
  }
  
  return(
    <DisplayContainer>
      { displayBooks }
    </DisplayContainer>
  )
}

export default DisplayResults