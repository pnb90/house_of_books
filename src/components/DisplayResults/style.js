import styled from 'styled-components'

export const DisplayContainer = styled.div`
  display: flex;
  width: calc(100vw - 15rem); 
  height: calc(100vh - 12rem);
  background: white;
  border-radius: 5px;
  flex-wrap: wrap;
  justify-content: center;
  padding: 1rem;
  overflow: auto;
  background: #f1f8e9;
`