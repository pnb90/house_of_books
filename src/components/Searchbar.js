import React, {useState} from 'react'
import styled from 'styled-components'
import InputBase from '@material-ui/core/InputBase'
import PropTypes from 'prop-types'

const Input = styled(InputBase)`
  background: white;
  border-radius: 5px;
  margin-top: 15px;
  margin-bottom: 15px;
  padding: 5px;
  background: #f1f8e9;
`

const InputLabel = styled.label`
  
`
function Searchbar(props) {
  const [searchTerm, setSearchTerm] = useState("")
  
  /** 
   * Updates the searchTerm to match the information being typed.
   * 
   * @param {event} - HTML event tracking information in the searchbar
   * 
   * Calls the setSearchTerm hook to update searchTerm upon any changes in the searchbar.
  */
 
  const handleSearchTermsChanges = (event) => { 
    setSearchTerm(event.target.value); 
  }
    
   /** 
   * Handles the submit action once entering the search term
   * 
   * @param {event} - HTML event tracking information in the searchbar
   * 
   * Prevents default behavior of the event, namely reloading the page.
   * Passes the searchTerm back to the parent component via the props.searchBookCallback function after downcasing it. 
   * 
  */
  const handleSubmit = (event) => {
    event.preventDefault()
    props.searchBookCallback( searchTerm.toLowerCase() )
  }

  return(
    <form
      onSubmit = { handleSubmit }
    >   
      <InputLabel htmlFor = "searchInput"> Book Search: </InputLabel> 
      <Input
        id = "searchInput"
        type = "text"
        aria-label = "book searchbar"
        onChange = { handleSearchTermsChanges }
        placeholder = "Search for books"
      />
    </form>
  )
}

Searchbar.propTypes = {
  searchBookCallback: PropTypes.func
}

export default Searchbar