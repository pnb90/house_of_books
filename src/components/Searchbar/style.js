import styled from 'styled-components'
import InputBase from '@material-ui/core/InputBase'

export const Input = styled(InputBase)`
  background: white;
  border-radius: 5px;
  margin-top: 15px;
  margin-bottom: 15px;
  padding: 5px;
  background: #f1f8e9;
`

export const InputLabel = styled.label`
  
`