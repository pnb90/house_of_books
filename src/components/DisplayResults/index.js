import React from 'react'
import Card from '../Card'
import PropTypes from 'prop-types'
import { DisplayContainer } from './style'

function DisplayResults(props) {
  var displayBooks = null

  if (props.books) {
    displayBooks = props.books.map(book => {
      return(
              <Card
                key = { book.id } 
                externalLink = { book.volumeInfo.infoLink }
                title = { book.volumeInfo.title }
                bookCover = { (book.volumeInfo.imageLinks && book.volumeInfo.imageLinks.thumbnail) || "https://images-na.ssl-images-amazon.com/images/I/91GQ%2BOWqgHL._SX425_.jpg" }
                publisher = { book.volumeInfo.publisher || "N/A" }
                description = { book.volumeInfo.description || "No description available." }
                authors = { book.volumeInfo.authors || "N/A" }
              />      
      )})
  } else if (props.totalItems === 0 ){
    displayBooks = "Oh no - 0 search results. Try another search"
  }
  
  return(
    <DisplayContainer>
      { displayBooks }
    </DisplayContainer>
  )
}

DisplayContainer.propTypes = {
  books: PropTypes.array,
  totalItems: PropTypes.number
}


export default DisplayResults