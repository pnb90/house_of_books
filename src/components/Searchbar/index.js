import React, {useState} from 'react'
import PropTypes from 'prop-types'
import { Input, InputLabel } from './style'

export default function Searchbar(props) {
  const [searchTerm, setSearchTerm] = useState("")
  
  /** 
   * Updates the searchTerm to match the information being typed.
   * 
   * @param {event} - HTML event tracking information in the searchbar
   * 
   * Calls the setSearchTerm hook to update searchTerm upon any changes in the searchbar.
  */
 
  const handleSearchTermsChanges = (event) => { 
    setSearchTerm(event.target.value); 
  }
    
   /** 
   * Handles the submit action once entering the search term
   * 
   * @param {event} - HTML event tracking information in the searchbar
   * 
   * Prevents default behavior of the event, namely reloading the page.
   * Passes the searchTerm back to the parent component via the props.searchBookCallback function after downcasing it. 
   * 
  */
  const handleSubmit = (event) => {
    event.preventDefault()
    props.searchBookCallback( searchTerm.toLowerCase() )
  }

  return(
    <form
      onSubmit = { handleSubmit }
      data-testid = "searchbar testID"
    >   
      <InputLabel htmlFor = "searchInput"> 
        Book Search: 
      </InputLabel> 
      <Input
        id = "searchInput"
        type = "text"
        aria-label = "book searchbar"
        onChange = { handleSearchTermsChanges }
        placeholder = "Search for books"
      />
    </form>
  )
}

Searchbar.propTypes = {
  searchBookCallback: PropTypes.func
}