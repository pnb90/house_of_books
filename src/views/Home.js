import React, {useState} from 'react'
import styled from 'styled-components'
import Searchbar from '../components/Searchbar'
import DisplayResults from '../components/DisplayResults'
import LocalLibrary from '@material-ui/icons/LocalLibrary'
import axios from 'axios'

// https://css-tricks.com/snippets/css/a-guide-to-flexbox/
const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  align-content: center;
  height: auto;
  width: auto;
`

const SiteName = styled.div`
  font-size: 46px;
  margin: 5px;
`

function Home() {
  
  // Retrieving the search terms from the searcbar 
  const [books, setBooks] = useState()
  const [totalItems, setTotalItems] = useState()

  const searchBookCallback = (dataFromChild) => {
    axios
      .get(`https://www.googleapis.com/books/v1/volumes?q=${dataFromChild}&maxResults=40`)
      .then(response => {
                          setTotalItems(response.data.totalItems)
                          setBooks(response.data.items)
                          })
      .catch(error => console.log(error))
  }

  return(
    <Container>
      <SiteName>
        <LocalLibrary />
          House of Books
        <LocalLibrary />
      </SiteName>

      <Searchbar 
        searchBookCallback = { searchBookCallback }
      />

      <DisplayResults 
        books = { books }
        totalItems = { totalItems }
      />

    </Container>
  )
}

export default Home