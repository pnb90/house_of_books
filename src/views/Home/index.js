import React, {useState} from 'react'
import Searchbar from 'components/Searchbar'
import DisplayResults from 'components/DisplayResults'
import LocalLibrary from '@material-ui/icons/LocalLibrary'
import axios from 'axios'
import { Container, SiteName } from './style'

export default function Home() {
  
  const [books, setBooks] = useState()
  const [totalItems, setTotalItems] = useState()
  
  const searchBookCallback = (dataFromChild) => {
    axios
      .get(`https://www.googleapis.com/books/v1/volumes?q=${dataFromChild}&maxResults=40`)
      .then(response => {
                          setTotalItems(response.data.totalItems)
                          setBooks(response.data.items)
                          })
      .catch(error => console.log(error))
  }

  return(
    <Container>
      <SiteName>
        <LocalLibrary />
          House of Books
        <LocalLibrary />
      </SiteName>

      <Searchbar
        searchBookCallback = { searchBookCallback }
      />

      <DisplayResults 
        books = { books }
        totalItems = { totalItems }
      />

    </Container>
  )
}