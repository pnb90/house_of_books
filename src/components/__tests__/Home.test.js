import React from 'react';
import { render, fireEvent, prettyDOM } from '@testing-library/react'
import "jest-dom/extend-expect";
import axiosMock from "axios";
import Home from 'views/Home';
import Searchbar from 'components/Searchbar'
import DisplayResults from 'components/DisplayResults'

// Tests I want to write: axios call works, receives props from searchbar, containers searchbar and displayresults components, passes information to both accurately. 
describe('Homepage component', () => {  
   it("matches snapshot", () => {
    const { asFragment } = render(<Home />)
    expect(asFragment()).toMatchSnapshot()
   })

   
   it("contains the Searchbar Component", () => {
      const { getByTestId } = render(<Home />)
      expect(getByTestId('searchbar testID')).toBeTruthy()
   })
   
   it("contains the DisplayResults Component", () => {
      const { getByTestId } = render(<Home />)
      expect(getByTestId('displayContainer testID')).toBeTruthy() 
   })

         // it("receives props properly", () => {
         //    const { } = render(<Home />)
         // })

   // it("fetches book data", () => {
   //    axiosMock.get({ data: { books: "books" }})
   //    const searchTerm = "pizza"
   //    const { getByTestId } = render(<Home />)
      
   //    expect(axiosMock.get).toHaveBeenCalled()
      
   //    expect(axiosMock.get).toHaveBeenCalledWith(url)
   // })

   });
