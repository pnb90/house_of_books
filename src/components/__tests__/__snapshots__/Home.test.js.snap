// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-fjdhpX bDUtko"
  >
    <div
      class="sc-jzJRlG ivLtnB"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 11.55C9.64 9.35 6.48 8 3 8v11c3.48 0 6.64 1.35 9 3.55 2.36-2.19 5.52-3.55 9-3.55V8c-3.48 0-6.64 1.35-9 3.55zM12 8c1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3 1.34 3 3 3z"
        />
      </svg>
      House of Books
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 11.55C9.64 9.35 6.48 8 3 8v11c3.48 0 6.64 1.35 9 3.55 2.36-2.19 5.52-3.55 9-3.55V8c-3.48 0-6.64 1.35-9 3.55zM12 8c1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3 1.34 3 3 3z"
        />
      </svg>
    </div>
    <form
      data-testid="searchbar testID"
    >
      <input
        aria-label="book searchbar"
        class="sc-bdVaJa fcBMVU"
        data-testid="input testID"
        id="searchInput"
        placeholder="Search for books"
        type="text"
      />
    </form>
    <div
      class="sc-jTzLTM kwYROD"
      data-testid="displayContainer testID"
    />
  </div>
</DocumentFragment>
`;
